receiver/dockerstatsreceiver/receiver.go

```diff
@@ -39,24 +41,15 @@ type receiver struct {
 	config   *Config
 	settings component.ReceiverCreateSettings
 	client   *docker.Client
+	mb       *metadata.MetricsBuilder
 }
 
-func NewReceiver(
-	_ context.Context,
-	set component.ReceiverCreateSettings,
-	config *Config,
-	nextConsumer consumer.Metrics,
-) (component.MetricsReceiver, error) {
-	recv := receiver{
+func newReceiver(set component.ReceiverCreateSettings, config *Config) *receiver {
+	return &receiver{
 		config:   config,
 		settings: set,
+		mb:       metadata.NewMetricsBuilder(config.MetricsConfig, set.BuildInfo),
```

Comment:
Can we propagate `ProvidePerCoreCPUMetrics` to the underlying per core cpu metrics enabled flag?

Reply:
Yeah, good idea. I've made it so if the user has that flag set to `true`, I override it on the mdatagen metric settings (which are false by default).
```go
if config.ProvidePerCoreCPUMetrics {
  config.MetricsConfig.ContainerCPUUsagePercpu.Enabled = config.ProvidePerCoreCPUMetrics
}
``` 

I've also enabled the per-cpu metrics in the test, so this behaviour is verified.